api/v1alpha1/applicationset_types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import (
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ApplicationSet is a set of Application resources
+// +kubebuilder:object:root=true
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ApplicationSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              ApplicationSetSpec `json:"spec"`
+}
+
+// ApplicationSetSpec represents a class of application set state.
+type ApplicationSetSpec struct {
+	Generators ApplicationSetGenerators  `json:"generators"`
+	Template   ApplicationSetTemplate    `json:"template"`
+	SyncPolicy *ApplicationSetSyncPolicy `json:"syncPolicy,omitempty"`
+}
+
+// ApplicationSetSyncPolicy will provide a syncPolicy similar to Applications
+type ApplicationSetSyncPolicy struct {
+	// Automated will keep an application synced to the target revision
+	Automated *SyncPolicyAutomated `json:"automated,omitempty"`
+}
+
+// SyncPolicyAutomated
+type SyncPolicyAutomated struct {
+	// Prune will prune resources automatically as part of automated sync (default: false)
+	Prune bool `json:"prune,omitempty"`
+	// Initial will perform an initial sync for any newly created Applications which do not have automated sync turned on.
+	Initial bool `json:"initial,omitempty"`
+}
+
+// ApplicationSetTemplate represents argocd ApplicationSpec
+type ApplicationSetTemplate struct {
+	metav1.ObjectMeta `json:"metadata"`
+	TemplateSpec      v1alpha1.ApplicationSpec `json:"spec"`
+}
+
+// ApplicationSetGenerators include list item info
+type ApplicationSetGenerators struct {
```

Comment:
We should remove `s` from ApplicationSetGenerators since this should be a single generator

```golang
type ApplicationSetGenerator struct {
	List *ListGenerator `json:"list, omitempty"`
}
```

Also, we should remove s from `ListGenerators` as well.